packages/template-retail-react-app/app/pages/registration/index.jsx

```diff
@@ -7,36 +7,53 @@
 
 import React, {useEffect} from 'react'
 import PropTypes from 'prop-types'
+import {useIntl} from 'react-intl'
 import {Box, Container} from '@chakra-ui/react'
-import useCustomer from '../../commerce-api/hooks/useCustomer'
-import Seo from '../../components/seo'
+import {
+    ShopperLoginHelpers,
+    useShopperLoginHelper,
+    useCustomerType
+} from 'commerce-sdk-react-preview'
 import {useForm} from 'react-hook-form'
+import {useLocation} from 'react-router-dom'
+import Seo from '../../components/seo'
 import RegisterForm from '../../components/register'
 import useNavigation from '../../hooks/use-navigation'
 import useEinstein from '../../commerce-api/hooks/useEinstein'
-import {useLocation} from 'react-router-dom'
+import {API_ERROR_MESSAGE} from '../../constants'
 
 const Registration = () => {
+    const {formatMessage} = useIntl()
     const navigate = useNavigation()
-    const customer = useCustomer()
+    const {isRegistered} = useCustomerType()
     const form = useForm()
     const einstein = useEinstein()
     const {pathname} = useLocation()
+    const register = useShopperLoginHelper(ShopperLoginHelpers.Register)
 
-    const submitForm = async (data) => {
-        try {
-            await customer.registerCustomer(data)
-        } catch (error) {
-            form.setError('global', {type: 'manual', message: error.message})
+    const submitForm = (data) => {
+        const body = {
+            customer: {
+                firstName: data.firstName,
+                lastName: data.lastName,
+                email: data.email,
+                login: data.email
+            },
+            password: data.password
```

Comment:
Note for possible future enhancement. React form hook allows you to name you inputs like `customer.firstName, customer.lastName, customer.email, customer,login, password` so when you get your data it's already in the correct structure.